scipy/optimize/_qap.py

```diff
@@ -0,0 +1,780 @@
+            returned.
+
+    Notes
+    -----
+    The algorithm may be sensitive to the initial permutation matrix (or
+    search "position") due to the possibility of several local minima
+    within the feasible region. A barycenter initialization is more likely to
+    result in a better solution than a single random initialization. However,
+    use of several randomized initializations  (through `init_weight` and
+    `init_k`) will likely result in a better solution at the cost of higher
+    runtime.
+
+    References
+    ----------
+    .. [1] J.T. Vogelstein, J.M. Conroy, V. Lyzinski, L.J. Podrazik,
+           S.G. Kratzer, E.T. Harley, D.E. Fishkind, R.J. Vogelstein, and
+           C.E. Priebe, "Fast approximate quadratic programming for graph
+           matching," PLOS one, vol. 10, no. 4, p. e0121002, 2015,
+           https://doi.org/10.1371/journal.pone.0121002
+
+    .. [2] D. Fishkind, S. Adali, H. Patsolic, L. Meng, D. Singh, V. Lyzinski,
+           C. Priebe, "Seeded graph matching", Pattern Recognit. 87 (2019):
+           203-215, https://doi.org/10.1016/j.patcog.2018.09.014
+    """
+
+    _check_unknown_options(unknown_options)
+
+    init_k = operator.index(init_k)
+    if init_k > 1 and init_weight is None:
+        init_weight = 0
+    maxiter = operator.index(maxiter)
+
+    # ValueError check
+    A, B, partial_match = _common_input_validation(A, B, partial_match)
+
+    msg = None
+    if (A < 0).any() or (B < 0).any():
+        msg = "`A` and `B` matrices must contain only non-negative elements."
+    elif isinstance(init_J, str) and init_J not in {'barycenter'}:
+        msg = "Invalid 'init_J' parameter string"
+    elif init_weight is not None and (init_weight < 0 or init_weight > 1):
+        msg = "'init_weight' must be strictly between zero and one"
+    elif init_k <= 0:
+        msg = "'init_k' must be a positive integer"
+    elif maxiter <= 0:
+        msg = "'maxiter' must be a positive integer"
+    elif eps <= 0:
+        msg = "'eps' must be a positive float"
+    if msg is not None:
+        raise ValueError(msg)
+
+    rng = check_random_state(rng)
+    n = A.shape[0]  # number of vertices in graphs
+    n_seeds = partial_match.shape[0]  # number of seeds
+    n_unseed = n - n_seeds
+
+    perm_inds = np.zeros(n)
+
+    obj_func_scalar = 1
+    if maximize:
+        obj_func_scalar = -1
+    best_score = obj_func_scalar * np.inf
+
+    seed_dist_c = np.setdiff1d(range(n), partial_match[:, 1])
+    if shuffle_input:
+        seed_dist_c = rng.permutation(seed_dist_c)
+        # shuffle_input to avoid results from inputs that were already matched
+
+    seed_cost_c = np.setdiff1d(range(n), partial_match[:, 0])
+    permutation_cost = np.concatenate([partial_match[:, 0],
+                                       seed_cost_c], axis=None).astype(int)
+    permutation_dist = np.concatenate([partial_match[:, 1],
+                                       seed_dist_c], axis=None).astype(int)
```

Comment:
Rename `permutation_cost` and `permutation_dist` to `perm_A` and `perm_B`.

Reply:
Done.